dir.traversal.targets

```diff
@@ -1,93 +1,68 @@
 
-  <Target Name="CleanAllProjects">
+  <Target Name="CleanAllProperties">
     <PropertyGroup>
       <DefaultCleanAllTarget Condition="'$(DefaultCleanAllTarget)'==''">Clean</DefaultCleanAllTarget>
+      <IterateProjectsTargets>$(DefaultCleanAllTarget)</IterateProjectsTargets>
+      <IterateProjectsProperties>CleanAllProjects=true</IterateProjectsProperties>
     </PropertyGroup>
-
-    <!-- To Serialize we use msbuild's batching functionality '%' to force it to batch all similar projects with the same identity 
-         however since the project names are unique it will essentially force each to run in its own batch -->
-    <MSBuild Targets="$(DefaultCleanAllTarget)"
-             Projects="@(Project)"
-             Condition="'$(SerializeProjects)'=='true'"
-             Properties="Dummy=%(Identity);CleanAllProjects=true"
-             ContinueOnError="ErrorAndContinue" />
-
-    <MSBuild Targets="$(DefaultCleanAllTarget)"
-             Projects="@(Project)"
-             Condition="'$(SerializeProjects)'!='true'"
-             Properties="CleanAllProjects=true"
-             BuildInParallel="true"
-             ContinueOnError="ErrorAndContinue" />
-
-    <!-- Given we ErrorAndContinue we need to propagate the error if the overall task failed -->
-    <Error Condition="'$(MSBuildLastTaskResult)'=='false'" />
   </Target>
 
-  <Target Name="RestoreAllProjectPackages">
+  <Target Name="RestoreAllProperties">
     <Message Importance="High" Text="Restoring project packages..." />
 
     <PropertyGroup>
       <DefaultRestoreAllPackagesTarget Condition="'$(DefaultRestoreAllPackagesTarget)'==''">RestorePackages</DefaultRestoreAllPackagesTarget>
+      <IterateProjectsTargets>$(DefaultRestoreAllPackagesTarget)</IterateProjectsTargets>
+      <IterateProjectsProperties>RestoreAllPackages=true</IterateProjectsProperties>
     </PropertyGroup>
-
-    <!-- To Serialize we use msbuild's batching functionality '%' to force it to batch all similar projects with the same identity 
-      however since the project names are unique it will essentially force each to run in its own batch -->
-    <MSBuild Targets="$(DefaultRestoreAllPackagesTarget)"
-             Projects="@(Project)"
-             Condition="'$(SerializeProjects)'=='true'"
-             Properties="Dummy=%(Identity);RestoreAllPackages=true"
-             ContinueOnError="ErrorAndContinue" />
-
-    <MSBuild Targets="$(DefaultRestoreAllPackagesTarget)"
-             Projects="@(Project)"
-             Condition="'$(SerializeProjects)'!='true'"
-             Properties="RestoreAllPackages=true"
-             BuildInParallel="true"
-             ContinueOnError="ErrorAndContinue" />
-
-    <!-- Given we ErrorAndContinue we need to propagate the error if the overall task failed -->
-    <Error Condition="'$(MSBuildLastTaskResult)'=='false'" />
   </Target>
 
   <PropertyGroup>
     <TraversalBuildDependsOn>
-      BuildAllProjects;
+      BuildAllProperties;
+      IterateProjects;
```

Comment:
If someone does a rebuild which will trigger all of these targets I don't believe IterateProjects will get called for anything beyond the first one that gets triggered.